include/grpc/grpc.h

```diff
@@ -143,6 +143,17 @@ GRPCAPI void grpc_completion_queue_shutdown(grpc_completion_queue *cq);
     drained and no threads are executing grpc_completion_queue_next */
 GRPCAPI void grpc_completion_queue_destroy(grpc_completion_queue *cq);
 
+/*********** EXPERIMENTAL API ************/
+/** Initializes a thread local cache for this cq
+  * grpc_flush_cq_tls_cache() MUST be called on the same thread.
+  */
+GRPCAPI void grpc_prepare_cq_tls_cache(grpc_completion_queue *cq);
```

Comment:
grpc_cq_tls_cache_prepare to be more consistent with naming throughout the core API (object name first, then action verb). Additionally, since this is API, I would suggest not using "tls" because of abbreviation-ambiguity. (Note that C++ style guide doesn't like abbreviations.) Please call it grpc_cq_thread_local_cache or grpc_thread_local_cq or something like that

Reply:
Additionally, even though these are experimental, please document the parameters and return values (more important for flush than prepare)